### PR TITLE
feat(internal): add step.Environment() function

### PR DESCRIPTION
**Pull request by jbrockopp**
Part of the effort for https://github.com/go-vela/community/issues/54

This adds a new `step.Environment()` function to the `go-vela/pkg-executor/internal/step` package.

The intention of this function is to setup the step container with all environment variables from the library resources.

https://github.com/go-vela/pkg-executor/blob/e185d399e2dbd388f8ae61dbee81d9311e1e210e/internal/step/environment.go#L15-L66

### CODE REVIEW

File: internal/step/environment.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+			channel = constants.DefaultRoute
+		}
+
+		// check if the workspace exists in the environment
+		workspace, ok := c.Environment["VELA_WORKSPACE"]
+		if !ok {
+			// set default for workspace
+			//
+			// TODO: consider making this a constant
+			workspace = "/vela"
```
Comment (wass3r) [inline internal/step/environment.go:38]: Default is not this anymore, right?

Comment (wass3r) [reply internal/step/environment.go:38]: Or it's just a fallback?

**Review by wass3r — APPROVED** [review]

File: internal/step/environment.go (hunk 1 above)
Comment (jbrockopp) [reply internal/step/environment.go:38]: That will be addressed once https://github.com/go-vela/types/pull/135 is pulled into this repo 👍

**Review by wass3r — APPROVED** [review]

**Review by kneal — APPROVED** [review]
LGTM 🐬

**Pull request merged**